### ISSUE TITLE
Review literature on Machine Learning algorithms for 3D aerodynamic predictions

### ISSUE THREAD

**Issue by antonellopaolino** [ami-iit/element_ml-for-aerodynamics#2]
I open this issue to collect the most relevant works on the fluid flow images analysis via AEs, subsequently I'll try to highlight the benefits and the drawbacks of using such algorithms for iRonCub geometry. 

cc @Giulero @gabrielenava @DanielePucci

**Comment by antonellopaolino** [comment]
## Towards scalable surrogate models based on Neural Fields for large scale aerodynamic simulations

**Authors:** Giovanni Catalani, Jean Fesquet, Xavier Bertrand, Frédéric Tost, Michael Bauerheim, Joseph Morlier.

**pdf:** https://arxiv.org/pdf/2505.14704

**code:** https://github.com/giovannicatalani/MARIO

The paper is based on what already presented in [Neural fields for rapid aircraft aerodynamics simulations](https://www.nature.com/articles/s41598-024-76983-w.pdf), published last year in Nature Scientific Reports.


### Abstract

This paper introduces a novel surrogate modeling framework for aerodynamic applications based on Neural Fields. The proposed approach, MARIO (Modulated Aerodynamic Resolution Invariant Operator), addresses non parametric geometric variability through an efficient shape encoding mechanism and exploits the discretization-invariant nature of Neural Fields. It enables training on significantly downsampled meshes, while maintaining consistent accuracy during full-resolution inference. These properties allow for efficient modeling of diverse flow conditions, while reducing computational cost and memory requirements compared to traditional CFD solvers and existing surrogate methods. The framework is validated on two complementary datasets that reflect industrial constraints. First, the AirfRANS dataset consists in a two-dimensional airfoil benchmark with non-parametric shape variations. Performance evaluation of MARIO on this case demonstrates an order of magnitude improvement in prediction accuracy over existing methods across velocity, pressure, and turbulent viscosity fields, while accurately capturing boundary layer phenomena and aerodynamic coefficients. Second, the NASA Common Research Model features three-dimensional pressure distributions on a full aircraft surface mesh, with parametric control surface deflections. This configuration confirms MARIO’s accuracy and scalability. Benchmarking against state-of-the-art methods demonstrates that Neural Field surrogates can provide rapid and accurate aerodynamic predictions under the computational and data limitations characteristic of industrial applications.


### Contents

This paper (and its framework already presented in the Nature paper) is strongly relevant for the problem of learning the 3D aerodynamic QoI distributions for different flow conditions. In particular, I became familiar with the term Implicit Neural Representation (INR), or also called Neural Fields, which are "neural networks parametrized by weight $\theta$ mapping points $x$ in Euclidean space to vector or scalar quantities:

$$
f_{\theta} (x): \mathbb{R}^d \rightarrow \mathbb{R}^{d_u}
$$

INRs can be thought of as a continuous approximation of an underlying signal, for which a discretely sampled version, is available for learning." Basically INRs are what I'm trying to learn on iRonCub aerodynamics (I just wasn't familiar with the term, but the mathematics is the same.

This paper presents a methodology which allows to learn an INR for aircraft pressure distributions. The interesting bit is that they separate the "global conditions" as the Mach number and the AoA, from the Eulerian space coordinates, which is something I was seeking to apply, the refined methodology (available in the following work: [Towards scalable surrogate models based on Neural Fields for large scale aerodynamic simulations](https://arxiv.org/pdf/2505.14704)) consists of mapping the eulerian space values (positions and other filed values) via a Fourier Feature Embedding $\gamma$ and then feed them in a MLP. At each layer of the MLP, a _modulation shift_ $\phi_l$ is added as additional bias vector to account for the global condition changes in the flow. This modulation shift is mapped from the global conditions $z$ via another MLP.

<img width="1158" height="849" alt="Image" src="https://github.com/user-attachments/assets/1f7abc29-e4cc-435c-831d-235432d45c7f" />

Therefore the mathematical expression of the method is:

$$
f_{\theta,\phi} (x) = W_L (\eta_{L-1} \circ \eta_{L-2} \circ \dots \circ \eta_{1} \circ \gamma(x) ) + b_{L}
$$

$$
\eta_{l} (\cdot) = ReLU ( W_l(\cdot) + b_l + \phi_{l} (z) )
$$

$$
\phi_{l} (z) = [h_{\psi} (z)]_{l} \in \mathbb{R}^{d_{l}}
$$

Therefore the output vector of the modulation shift MLP is the concatenation of all the layers modulations.

### Comments

* The results seem interesting, in particular I would like to explore the addition of the Fourier Feature Encoding which seems to provide benefits in reconstrcting higher-frquency target data (see [Fourier Features Let Networks Learn High Frequency Functions in Low Dimensional Domains](https://proceedings.neurips.cc/paper_files/paper/2020/file/55053683268957697aa39fba6f231c68-Paper.pdf)) especially when dealing with INRs;
* Another intersting point is the separation between local input $x$ and geometric/flow input $z$ which are then recombined after some processing in the main MLP, this could be a valid alternative to feeding both of them for each sample;

cc @Giulero @DanielePucci

**Comment by antonellopaolino** [comment]
Closing for the moment, we reached a good understanding of the current state-of-the-art related to INRs.
